### PR TITLE
Fixes deprecation warnings introduced in React 15.5.0

### LINKED ISSUE #114: React 15.5 Deprecation Warnings Appearing Due to React-Bootstrap-Date-Picker import.
On application load, my bundle shoots out a warning about using both React.PropTypes and React.createClass. I traced it back to an import on one of my pages that use this very nice date picker component. The component eventually uses the "Portal" component which is the actual culprit that uses React.createClass.

I was wondering when this is going to get updated to get rid of these deprecation issues.

Thanks,

Jaron

* Version used:
* Browser Name and version: Chrome 57.0.2987.133
* Operating System and version (desktop or mobile): Win 7
deprecationscreenshot

### PR DESCRIPTION
**Pull request by charliecochran**
## Description
* Updates dependencies
* React.PropTypes -> PropTypes
* React.createClass -> createReactClass

## Motivation and Context
React 15.5.0 introduced deprecation warnings for PropTypes, createClass, and react-addons: https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html

Fixes #114 

## How Has This Been Tested?
Ran npm test on my machine, all tests passed.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by pinkstrings** [comment]
This pull request has been up for a month -- any word on when we can get this merged in?

File: example/app.jsx [hunk 1]
```diff
@@ -11,13 +11,14 @@ import Glyphicon from 'react-bootstrap/lib/Glyphicon';
...
 import HelpBlock from 'react-bootstrap/lib/HelpBlock';
+import createReactClass from 'create-react-class';
 
 const spanishDayLabels = ['Dom', 'Lu', 'Ma', 'Mx', 'Ju', 'Vi', 'Sab'];
 const spanishMonthLabels = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'];
 const wapperDivStyle = { border: '1px solid #ccc' };
 const scrollingDivStyle = { padding: '10px', height: '70px', overflow: 'auto' };
 
-const App = React.createClass({
+const App = createReactClass({
```
Comment (nZeus) [inline example/app.jsx]: As far as I understand, `React.createClass` can be replaced with `React.Component` and then there is no need to have a dependency on a `create-react-class` library. 
That's how another library did the same fix: https://github.com/Hacker0x01/react-datepicker/pull/822/files#diff-04c6e90faac2675aa89e2176d2eec7d8L34

**Comment by rubancar** [comment]
I'm currently using this library but I don't like to see those warnings and this pull request seems to be well-structured, why don't we get this update?

**Comment by tonoslav** [comment]
This got merged and after reinstall from this repo I keep getting the same error, any idea why?

**Comment by kunyan** [comment]
@tonoslav we have to wait a new release.

**Comment by jeinz11** [comment]
any updates on this? warnings are still here

**Pull request merged**